### PR TITLE
[v6r19] Add deprecated decorator

### PR DESCRIPTION
**Pull request by sfayer**
Hi,

Here is my suggestion for adding a simple deprecated decorator (for issue #3518)... I was initially going to do something more complicated but I ran into quite a few minor implementation details of decorators, so kept it simpler in the end.

Regards,
Simon

BEGINRELEASENOTES
*Core
NEW: Add deprecated decorator
ENDRELEASENOTES

### CODE REVIEW

**Comment by fstagni** [comment]
Thanks for this, it looks good to me. We can straight away apply it, for example for Job.py `setParametricInputSandbox` and to `setParametricInputData`.  Any objections?

**Comment by chaen** [comment]
I have a few comments

* I would be in favor of using a simple print rather than gLogger not to have loop dependency again
* Have you tried to apply it to several functions ? From a quick glance, I am afraid it will not work as expected. But I might be very wrong.

File: Core/Utilities/Decorators.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        If set, the deprecation warning will only be displayed on the first use.
+
+      Returns
+      -------
+      function
+        A double-function wrapper around the decorated object as required by the python
+        interpreter.
+  """
+
+  def decFunc( func, clsName=None ):
```
Comment (chaen) [inline Core/Utilities/Decorators.py]: How and where is clsName set ?

**Comment by chaen** [comment]
Also the arguments are not returned correctly for the doc

```
@deprecated("dont know")
def func2(a = 1):
  """ heeeeelp 
      
      :param a: is 1
  """
  pass
```

Result in
```
Help on function func2 in module __main__:

func2(*args, **kwargs)
    heeeeelp 
    
    :param a: is 1
```

Notice `*args, **kwargs`

**Comment by chaen** [comment]
About my worry with several function, forget it, more careful analysis reassured me :-)

File: Core/Utilities/Decorators.py (hunk 1 above)
Comment (sfayer) [reply Core/Utilities/Decorators.py]: It's used on line 87, I've added a patch to make it a little bit clearer.

**Comment by sfayer** [comment]
Hi,

I've been looking fixing the function signatures in the help, but it appears this isn't very easy on python2 (python3.3 added a __signature__ attribute to make this easy). The only way I've found of doing this in 2.7 is to generate another outer wrapper using lamba (and eval) with a matching signature, but that increases the complexity quite significantly and is likely to be more fragile. I can do that if you really want, but thought I'd see what you think before putting too much effort into it...

Regards,
Simon

File: Core/Utilities/Decorators.py (hunk 1 above)
Comment (chaen) [reply Core/Utilities/Decorators.py]: Yes but how do you actually pass it ?

**Comment by chaen** [comment]
yeah your'e right. In any case, since it is for deprecated method, we do not really care ...

**Pull request merged**